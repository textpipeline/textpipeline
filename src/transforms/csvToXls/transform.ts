import { Transform } from '../types';

const csvToXls: Transform = {
  name: 'CSV to XLS',
  slug: 'csv-to-xls',
  inputType: 'CSV',
  outputType: 'XLS',
  project: {
    name: 'xlsx',
    projectHref: 'https://sheetjs.com/opensource',
    repositoryHref: 'https://github.com/SheetJS/js-xlsx',
    license: 'apache-2.0',
  },
  execute: input => Promise.resolve(`TEST OUTPUT: ${input}`),
};

export default csvToXls;
