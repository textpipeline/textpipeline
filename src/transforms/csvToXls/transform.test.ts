import csvToXls from './transform';

describe('CSV to XLS', () => {
  it('has the correct attributes', () => {
    expect(csvToXls).toMatchSnapshot();
  });

  describe('when called with valid input', () => {
    let output: string;

    beforeEach(async () => {
      output = await csvToXls.execute('stuff');
    });

    it('produces the correct output', () => {
      expect(output).toMatchSnapshot();
    });
  });
});
