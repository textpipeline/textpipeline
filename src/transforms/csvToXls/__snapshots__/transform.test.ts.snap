// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSV to XLS has the correct attributes 1`] = `
Object {
  "execute": [Function],
  "inputType": "CSV",
  "name": "CSV to XLS",
  "outputType": "XLS",
  "project": Object {
    "license": "apache-2.0",
    "name": "xlsx",
    "projectHref": "https://sheetjs.com/opensource",
    "repositoryHref": "https://github.com/SheetJS/js-xlsx",
  },
  "slug": "csv-to-xls",
}
`;

exports[`CSV to XLS when called with valid input produces the correct output 1`] = `"TEST OUTPUT: stuff"`;
